expression/src/main/java/io/smallrye/common/expression/Expression.java

```diff
@@ -551,7 +588,22 @@ private static Node parseString(Itr itr, final boolean allowExpr, final boolean
                     //throw Assert.unreachableCode();
                 }
                 case '\\': {
-                    if (flags.contains(Flag.ESCAPES)) {
+                    if (flags.contains(Flag.NO_$$)) {
+                        int escape = itr.getNextIdx();
+                        if (itr.hasNext() && itr.peekNext() == '$') {
+                            itr.next();
+                            if (itr.hasNext() && itr.peekNext() == '{') {
+                                list.add(new LiteralNode(itr.getStr(), start, escape - 1));
+                                list.add(new LiteralNode(itr.getStr(), escape, itr.getNextIdx()));
+                                start = itr.getNextIdx();
+                            } else if (itr.hasNext() && itr.peekNext() == '$') {
+                                list.add(new LiteralNode(itr.getStr(), start, escape));
+                                start = itr.getPrevIdx();
+                                itr.prev();
+                            }
+                        }
+                        continue;
```

Comment:
Why do we care about `NO_$$` here?

Reply:
The goal here was to support the MP escape sequence of `\$`, without turning on `ESCAPES`, which breaks commas separated values escapes.